### DISCUSSION TITLE
Antialiassing

### DISCUSSION THREAD

**Discussion by czarnyafgan** [Try/OpenGothic#192] [General]
Antialiassing feature with processing level 0/4/8 etc. and turning on/off configurable from main menu. 

Please, consider it in next release, it's so needed!

**Comment by Try** [comment]
Hi, @czarnyafgan !

Yep, I haven't forgot about antialiasing just thinking about in engine api for that in background :)
So, question to developers around here: how in engine api should look a like?

Current in engine api look like this:
```c++
// allocate frame buffer with 4 color attachments + zbuffer
fboGBuf     = device.frameBuffer(lightingBuf,gbufDiffuse,gbufNormal,gbufDepth,zbuffer);

// render-pass (need a better name for it), - controls Load/Store operations
gbufPass  = device.pass(FboMode(FboMode::PreserveOut,Color(0.0)),
                        FboMode(FboMode::PreserveOut),
                        FboMode(FboMode::PreserveOut),
                        FboMode(FboMode::PreserveOut,Color(1.0)),
                        FboMode(FboMode::PreserveOut,1.f));
```

Need an api to express:
1. Lazy allocated attachment, for MSAA texture
2. Resolve api (I think store op `FboMode::Resolve` should be fine)
3. While on that would be nice to cover shading-rate-attachments and pixel-density-map
